### PR TITLE
TINY-6102: Fix common eslint rules not enabled that were causing issues

### PR DESCRIPTION
**Pull request by lnewson**
Description of Changes:
* This enables the `keyword-spacing`, `prefer-const` and `comma-dangle` rules

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] License headers added on new files (if applicable)

Review:
* [x] Milestone set
* [ ] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Comment by lnewson** [comment]
Note: I found out the types in `Schema.ts` was missing certain properties and that the `DebounceFunc` type in `Delay.ts` was incorrect while doing this.

File: modules/tinymce/src/core/test/ts/browser/html/NodeTest.ts [hunk 1]
```diff
@@ -310,7 +284,7 @@ UnitTest.asynctest('browser.tinymce.core.html.NodeTest', function (success, fail
   });
 
   suite.test('clone', function () {
-    let root, node, clone;
+    let  node, clone;
```
Comment (ashfordneil) [inline modules/tinymce/src/core/test/ts/browser/html/NodeTest.ts:287]:
```ts
let  node
```

or

```ts
let node
```

Definitely a minor nit but also this seems like the PR for minor nits?

File: modules/tinymce/src/plugins/save/main/ts/core/Actions.ts [hunk 2]
```diff
@@ -37,7 +36,9 @@ const save = function (editor) {
   if (formObj) {
     editor.setDirty(false);
 
-    if (!formObj.onsubmit || formObj.onsubmit()) {
+    // TODO: this is probably broken, as an event should be passed to `onsubmit`
```
Comment (ashfordneil) [inline modules/tinymce/src/plugins/save/main/ts/core/Actions.ts:39]: Is there a matched Jira?

File: modules/agar/src/test/ts/atomic/api/TestLogsTest.ts [hunk 3]
```diff
@@ -1,13 +1,6 @@
...
 import { Arr } from '@ephox/katamari';
-import {
-  addLogEntry,
-  addStackTrace,
-  popLogLevel,
-  pushLogLevel,
-  TestLogEntryState,
-  TestLogs
-} from 'ephox/agar/api/TestLogs';
+import { addLogEntry, addStackTrace, popLogLevel, pushLogLevel, TestLogEntryState, TestLogs } from 'ephox/agar/api/TestLogs';
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/agar/src/test/ts/atomic/api/TestLogsTest.ts:3]: Yes, we were going to limit it at 140 or 160 eventually.

File: modules/tinymce/src/core/main/ts/fmt/ApplyFormat.ts [hunk 4]
```diff
@@ -235,8 +233,8 @@ const applyFormat = function (ed: Editor, name: string, vars?: FormatVars, node?
         return count;
       };
 
-      const getChildElementNode = function (root: Node) {
-        let child: Node | boolean = false;
+      const getChildElementNode = function (root: Node): Node | false {
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/fmt/ApplyFormat.ts:236]: Yeah agreed, but I was trying to not change anything functionality wise.

File: modules/tinymce/src/core/test/ts/browser/html/NodeTest.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/core/test/ts/browser/html/NodeTest.ts:287]: Yeah, that's a bug. Thanks for the catch! I wonder why eslint didn't catch this...

File: modules/tinymce/src/plugins/save/main/ts/core/Actions.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/save/main/ts/core/Actions.ts:39]: @metricjs did log one, I'll see if I can dig it up.

**Pull request merged**